test/dhcp_isc/state_changes_observer_test.rb

```diff
@@ -94,14 +184,42 @@ def test_update_subnet_service_with_dhcp_records_should_delete_leases_with_dupli
     assert @service.find_lease_by_mac("192.168.0.0", "00:11:22:33:44:66")
   end
 
+  def test_event_loop_with_started_event
+    @observer.kill_event_loop
```

Comment:
We could, although there might be a slight slowdown associated with waiting for the worker thread to exit. I'll make the change to keep things easier to understand.